packages/paste-core/components/form/src/FormRadio.tsx

```diff
@@ -0,0 +1,139 @@
+  isDisabled?: boolean;
+  isInvalid?: boolean;
+}
+
+const FormRadio = React.forwardRef<HTMLInputElement, FormRadioProps>(
+  (
+    {
+      id,
+      name,
+      value,
+      'aria-label': ariaLabel,
+      'aria-labelledby': ariaLabelledBy,
+      defaultIsChecked,
+      isChecked,
+      isFullWidth,
+      isDisabled,
+      isInvalid,
+      onChange,
+      onBlur,
+      onFocus,
+      children,
+      ...props
+    },
+    ref
+  ) => {
+    let dotBackgroundColor = null as BackgroundColor;
+    if (isDisabled && isChecked) {
+      dotBackgroundColor = 'colorBackgroundInverse';
+    }
+
+    return (
+      <Box
+        {...safelySpreadBoxProps(props)}
+        as="label"
+        display="inline-flex"
+        verticalAlign="top"
+        htmlFor={id}
+        alignItems="center"
+        width={isFullWidth ? '100%' : undefined}
+        cursor={isDisabled ? 'not-allowed' : 'pointer'}
+      >
+        <ScreenReaderOnly
+          as="input"
+          // @ts-ignore
+          type="radio"
+          aria-label={ariaLabel}
+          aria-labelledby={ariaLabelledBy}
+          id={id}
+          ref={ref}
+          name={name}
+          value={value}
+          aria-invalid={isInvalid}
+          defaultChecked={defaultIsChecked}
+          onChange={onChange}
+          onBlur={onBlur}
+          onFocus={onFocus}
+          checked={isChecked}
+          disabled={isDisabled}
+        />
+        <SiblingBox
+          type="radio"
+          width="sizeIcon10"
+          height="sizeIcon10"
+          borderWidth="borderWidth10"
+          borderStyle="solid"
+          borderRadius="borderRadiusCircle"
+          borderColor="colorBorder"
+          _hover={{
+            borderColor: 'colorBorderPrimary',
+          }}
+          _focus={{
+            boxShadow: 'shadowFocus',
+          }}
+          // _active={{
+          //   borderColor: 'colorBorderPrimaryDarker',
+          //   backgroundColor: 'colorBackgroundPrimaryDarker',
+          // }}
+          _checked={{
+            borderColor: 'colorBorderPrimary',
+            backgroundColor: 'colorBackgroundPrimary',
+          }}
+          _disabled={{
+            borderColor: 'colorBorder',
+          }}
+          _invalid={{
+            borderColor: 'colorBorderError',
+          }}
+          _checkedAndHover={{
+            borderColor: 'colorBorderPrimaryDarker',
+            backgroundColor: 'colorBackgroundPrimaryDarker',
+          }}
+          // _checkedAndActive={{
+          //   borderColor: 'colorBorderPrimaryDarker',
+          //   backgroundColor: 'colorBackgroundPrimaryDarker',
+          // }}
+          _checkedAndDisabled={{
+            borderColor: 'colorBorder', // need new token
+            backgroundColor: 'colorBackground', // need new token
+          }}
+        >
+          <Box
+            as="span"
+            // @ts-ignore
+            width="0.375rem"
+            // @ts-ignore
+            height="0.375rem"
```

Comment:
Probably a design question as well. Not sure what to do about this width/height here.

Reply:
Resolved with the square size tokens.